lib/api/core/validation/types/integer.js

```diff
@@ -1,60 +1,81 @@
-var
-  util = require('util'),
-  BaseConstructor = require('../baseType'),
-  allowedRangeOptions = ['min', 'max'];
-
-/**
- * @constructor
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2017 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
  */
-function IntegerType () {
-  this.typeName = 'integer';
-  this.allowChildren = false;
-  this.allowedTypeOptions = ['range'];
-}
 
-util.inherits(IntegerType, BaseConstructor);
+const
+  BaseType = require('../baseType'),
+  allowedRangeOptions = ['min', 'max'];
 
 /**
- * @param {TypeOptions} typeOptions
- * @param {*} fieldValue
- * @param {string[]} errorMessages
+ * @class IntegerType
  */
-IntegerType.prototype.validate = function integerTypeValidate (typeOptions, fieldValue, errorMessages) {
-  if (typeof fieldValue !== 'number' || !Number.isInteger(fieldValue)) {
-    errorMessages.push('The field must be an integer.');
-    return false;
+class IntegerType extends BaseType {
+  constructor() {
+    super();
+    this.typeName = 'integer';
+    this.allowChildren = false;
+    this.allowedTypeOptions = ['range'];
   }
 
-  if (typeOptions.hasOwnProperty('range')) {
-    if (typeOptions.range.hasOwnProperty('min') && fieldValue < typeOptions.range.min) {
-      errorMessages.push('The value is lesser than the minimum.');
+  /**
+   * @param {TypeOptions} typeOptions
+   * @param {*} fieldValue
+   * @param {string[]} errorMessages
+   */
+  validate(typeOptions, fieldValue, errorMessages) {
+    if (typeof fieldValue !== 'number' || !Number.isInteger(fieldValue)) {
+      errorMessages.push('The field must be an integer.');
       return false;
     }
 
-    if (typeOptions.range.hasOwnProperty('max') && fieldValue > typeOptions.range.max) {
-      errorMessages.push('The value is greater than the maximum.');
-      return false;
+    if (typeOptions.hasOwnProperty('range')) {
+      if (typeOptions.range.hasOwnProperty('min') && fieldValue < typeOptions.range.min) {
+        errorMessages.push('The value is lesser than the minimum.');
+        return false;
+      }
+
+      if (typeOptions.range.hasOwnProperty('max') && fieldValue > typeOptions.range.max) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 90 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)